packages/repo-mock/src/metadata.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  Metadata,
+  MetaFile,
+  Root,
+  Snapshot,
+  TargetFile,
+  Targets,
+  Timestamp,
+} from '@tufjs/models';
+import { digestSHA256 } from './crypto';
+import { KeyPair } from './key';
+
+export function createTargetsMeta(
+  targetFiles: TargetFile[],
+  keyPair: KeyPair
+): Metadata<Targets> {
+  const targets = new Metadata(
+    new Targets({
+      expires: getExpires(),
+    })
+  );
+
+  targetFiles.forEach((targetFile) => targets.signed.addTarget(targetFile));
+
+  targets.sign((data) => keyPair.sign(data));
+
+  return targets;
+}
+
+export function createSnapshotMeta(
+  targets: Metadata<Targets>,
+  keyPair: KeyPair
+): Metadata<Snapshot> {
+  const snapshot = new Metadata(
+    new Snapshot({
+      expires: getExpires(),
+      meta: { 'targets.json': toMetaFile(targets) },
+    })
+  );
+
+  snapshot.sign((data) => keyPair.sign(data));
+  return snapshot;
+}
+
+export function createTimestampMeta(
+  snapshot: Metadata<Snapshot>,
+  keyPair: KeyPair
+): Metadata<Timestamp> {
+  const timestamp = new Metadata(
+    new Timestamp({
+      expires: getExpires(),
+      snapshotMeta: toMetaFile(snapshot),
+    })
+  );
+
+  timestamp.sign((data) => keyPair.sign(data));
+
+  return timestamp;
+}
+
+export function createRootMeta(keyPair: KeyPair): Metadata<Root> {
+  const root = new Metadata(
+    new Root({
+      expires: getExpires(),
+      consistentSnapshot: false,
+    })
+  );
+
+  root.signed.addKey(keyPair.publicKey, 'root');
+  root.signed.addKey(keyPair.publicKey, 'snapshot');
+  root.signed.addKey(keyPair.publicKey, 'targets');
+  root.signed.addKey(keyPair.publicKey, 'timestamp');
+
+  root.sign((data) => keyPair.sign(data));
+
+  return root;
+}
+
+function toMetaFile<T extends Root | Snapshot | Targets | Timestamp>(
+  metadata: Metadata<T>
+): MetaFile {
+  const meta = JSON.stringify(metadata);
+  return new MetaFile({
+    version: 1,
+    length: meta.length,
+    hashes: { sha256: digestSHA256(meta) },
+  });
+}
+
+function getExpires(): string {
+  const expires = new Date();
+  expires.setFullYear(expires.getFullYear() + 1);
+  return expires.toISOString();
```

Comment:
In tuf spec, all metadata's EXPIRES are the same as is described for the root.json file. Just wonder by calling the function, would that have problem that there is a little offset between each EXPIRES data in metadata and root

Reply:
I was just looking at the metadata files in the Sigstore TUF repo and they all have different expiration values . . . which sorta makes sense cause the root may be long-lived (expires 1 year from now) whereas the timestamp metadata should expire pretty quickly.

Reply:
that makes sense! I was a little bit confused by the spec..